### PR TITLE
container: Added Dockerfile and startup-all-in-one.sh

### PR DESCRIPTION
**Pull request by ashcrow**
To build:
```
    docker build -t commissaire .
```

To run:
```
    docker run -p 8000:8000 -d --name commissaire commissaire
```

### CODE REVIEW

**Review by mbarnes — COMMENTED** [review]
Still testing this out.  Code looks good to me but I wonder if these files should live under a subdirectory like `demo` or something, since this isn't the way we'd expect Commissaire to be run in production.

**Review by mbarnes — CHANGES_REQUESTED** [review]
The container stops immediately for me.  I substituted the `-d` with `-t` in the run command to see what's going on.  Here's the output.  Appears (to me) to be an issue with etcd configuration.
```
7:C 25 Jan 15:41:22.339 # Warning: no config file specified, using the default config. In order to specify a config file use redis-server /path/to/redis.conf
+ Creating Commissaire keyspaces...
++ Creating /commissaire/clusters
                _._                                                  
           _.-``__ ''-._                                             
      _.-``    `.  `_.  ''-._           Redis 3.0.6 (00000000/0) 64 bit
  .-`` .-```.  ```\/    _.,_ ''-._                                   
 (    '      ,       .-`  | `,    )     Running in standalone mode
 |`-._`-...-` __...-.``-._|'` _.-'|     Port: 6379
 |    `-._   `._    /     _.-'    |     PID: 7
  `-._    `-._  `-./  _.-'    _.-'                                   
 |`-._`-._    `-.__.-'    _.-'_.-'|                                  
 |    `-._`-._        _.-'_.-'    |           http://redis.io        
  `-._    `-._`-.__.-'_.-'    _.-'                                   
 |`-._`-._    `-.__.-'    _.-'_.-'|                                  
 |    `-._`-._        _.-'_.-'    |                                  
  `-._    `-._`-.__.-'_.-'    _.-'                                   
      `-._    `-.__.-'    _.-'                                       
          `-._        _.-'                                           
              `-.__.-'                                               

7:M 25 Jan 15:41:22.341 # WARNING: The TCP backlog setting of 511 cannot be enforced because /proc/sys/net/core/somaxconn is set to the lower value of 128.
7:M 25 Jan 15:41:22.341 # Server started, Redis version 3.0.6
7:M 25 Jan 15:41:22.341 # WARNING overcommit_memory is set to 0! Background save may fail under low memory condition. To fix this issue add 'vm.overcommit_memory = 1' to /etc/sysctl.conf and then reboot or run the command 'sysctl vm.overcommit_memory=1' for this to take effect.
7:M 25 Jan 15:41:22.341 # WARNING you have Transparent Huge Pages (THP) support enabled in your kernel. This will create latency and memory usage issues with Redis. To fix this issue run the command 'echo never > /sys/kernel/mm/transparent_hugepage/enabled' as root, and add it to your /etc/rc.local in order to retain the setting after a reboot. Redis must be restarted after THP is disabled.
7:M 25 Jan 15:41:22.341 * The server is now ready to accept connections on port 6379
2017-01-25 15:41:22.356761 I | etcdmain: etcd Version: 2.3.3
2017-01-25 15:41:22.356802 I | etcdmain: Git SHA: c41345d
2017-01-25 15:41:22.356809 I | etcdmain: Go Version: go1.6.3
2017-01-25 15:41:22.356813 I | etcdmain: Go OS/Arch: linux/amd64
2017-01-25 15:41:22.356819 I | etcdmain: setting maximum number of CPUs to 8, total number of available CPUs is 8
2017-01-25 15:41:22.356823 W | etcdmain: no data-dir provided, using default data-dir ./default.etcd
Error:  client: etcd cluster is unavailable or misconfigured
error #0: dial tcp 127.0.0.1:2379: getsockopt: connection refused
error #1: dial tcp 127.0.0.1:4001: getsockopt: connection refused

2017-01-25 15:41:22.357642 I | etcdmain: listening for peers on http://localhost:2380
2017-01-25 15:41:22.357794 I | etcdmain: listening for peers on http://localhost:7001
2017-01-25 15:41:22.357904 I | etcdmain: listening for client requests on http://localhost:2379
2017-01-25 15:41:22.358001 I | etcdmain: listening for client requests on http://localhost:4001
++ Creating /commissaire/cluster
2017-01-25 15:41:22.358300 I | etcdserver: name = default
2017-01-25 15:41:22.358308 I | etcdserver: data dir = default.etcd
2017-01-25 15:41:22.358318 I | etcdserver: member dir = default.etcd/member
2017-01-25 15:41:22.358329 I | etcdserver: heartbeat = 100ms
2017-01-25 15:41:22.358334 I | etcdserver: election = 1000ms
2017-01-25 15:41:22.358339 I | etcdserver: snapshot count = 10000
2017-01-25 15:41:22.358349 I | etcdserver: advertise client URLs = http://localhost:2379,http://localhost:4001
2017-01-25 15:41:22.358356 I | etcdserver: initial advertise peer URLs = http://localhost:2380,http://localhost:7001
2017-01-25 15:41:22.358366 I | etcdserver: initial cluster = default=http://localhost:2380,default=http://localhost:7001
2017-01-25 15:41:22.412285 I | etcdserver: starting member ce2a822cea30bfca in cluster 7e27652122e8b2ae
2017-01-25 15:41:22.412336 I | raft: ce2a822cea30bfca became follower at term 0
2017-01-25 15:41:22.412353 I | raft: newRaft ce2a822cea30bfca [peers: [], term: 0, commit: 0, applied: 0, lastindex: 0, lastterm: 0]
2017-01-25 15:41:22.412358 I | raft: ce2a822cea30bfca became follower at term 1
2017-01-25 15:41:22.412677 I | etcdserver: starting server... [version: 2.3.3, cluster version: to_be_decided]
2017-01-25 15:41:22.413536 N | etcdserver: added local member ce2a822cea30bfca [http://localhost:2380 http://localhost:7001] to cluster 7e27652122e8b2ae
etcd cluster has no published client endpoints.
Try '--no-sync' if you want to access non-published client endpoints(http://127.0.0.1:4001,http://127.0.0.1:2379).
Error:  client: no endpoints available
++ Creating /commissaire/hosts
etcd cluster has no published client endpoints.
Try '--no-sync' if you want to access non-published client endpoints(http://127.0.0.1:2379,http://127.0.0.1:4001).
Error:  client: no endpoints available
++ Creating /commissaire/networks
etcd cluster has no published client endpoints.
Try '--no-sync' if you want to access non-published client endpoints(http://127.0.0.1:4001,http://127.0.0.1:2379).
Error:  client: no endpoints available
++ Creating /commissaire/status
etcd cluster has no published client endpoints.
Try '--no-sync' if you want to access non-published client endpoints(http://127.0.0.1:2379,http://127.0.0.1:4001).
Error:  client: no endpoints available
+ Creating default network configuration...
etcd cluster has no published client endpoints.
Try '--no-sync' if you want to access non-published client endpoints(http://127.0.0.1:4001,http://127.0.0.1:2379).
Error:  client: no endpoints available
```

**Comment by ashcrow** [comment]
Interesting ... I just gave it another shot and this is what I got.

```
$ sudo docker run -p 8000:8000 -d --name localtest localtest
638e50605a45297199c9bbf8de2c8435970965ea7278527b5ae11cccf3f0ae6f
$ sudo docker ps
CONTAINER ID        IMAGE               COMMAND                  CREATED             STATUS              PORTS                    NAMES
638e50605a45        localtest           "/bin/sh -c /commissa"   5 seconds ago       Up 3 seconds        0.0.0.0:8000->8000/tcp   localtest
$ curl -u "a:a" http://127.0.0.1:8000/api/v0/networks/
["default"]$ 
```

Here is what I'm running with ... maybe it's a difference of versions ...

```
$ sudo docker images | grep fedora
docker.io/fedora              latest              a1e614f0f30e        5 weeks ago         197.1 MB
$ rpm -q docker
docker-1.10.3-55.gite03ddb8.fc24.x86_64
$ sudo docker run -ti --rm -p 8000:8000 localtest /bin/bash
[root@57182b66659d commissaire]# rpm -q etcd redis
etcd-3.0.15-1.fc25.x86_64
redis-3.2.3-1.fc25.x86_64
```

Does this look similar to your setup?

**Comment by mbarnes** [comment]
I have a newer `docker`.

Not sure why our `docker.io/fedora` images disagree if we're both on `latest`.  I even just rebuilt the image with `--no-cache`.
```
$ sudo docker images | grep fedora
docker.io/fedora                   latest              11a5107645d4        5 months ago        204.4 MB
$ rpm -q docker
docker-1.12.6-4.gitf499e8b.fc25.x86_64
$ rpm -q etcd redis
etcd-3.0.15-1.fc25.x86_64
redis-3.2.3-1.fc25.x86_64
```

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,37 @@
+FROM fedora
```
Comment (mbarnes) [inline Dockerfile]: Maybe specify a release?  `fedora:25` is more predictable than `fedora:latest`.

Comment (ashcrow) [reply Dockerfile]: Good idea. I'll update that.

**Comment by ashcrow** [comment]
@mbarnes I removed the base fedora image and my previous build and rebuilt the container. I ran it with:

```
$ sudo docker run -ti --rm -p 8000:8000 localtest /bin/bash
(in the container)$ ./startup-all-in-one.sh
```

and it worked fine still. Can you try that?

**Comment by mbarnes** [comment]
@ashcrow I did the same and it works now.  Also, my base fedora image ID now matches yours (`a1e614f0f30e`), so it must have been something about my old image.  Weird.

**Review by mbarnes — APPROVED** [review]

**Comment by mbarnes** [comment]
Merged manually.  Sorry for the confusion.

**Pull request closed without merging**